web3.js/src/keypair.ts

```diff
@@ -0,0 +1,76 @@
+import * as nacl from 'tweetnacl';
+import type {SignKeyPair} from 'tweetnacl';
+
+import {PublicKey} from './publickey';
+
+/**
+ * An account keypair used for signing transactions.
+ */
+export class Keypair {
+  /**
+   * @internal
+   *
+   * Create a new keypair instance from a {@link SignKeyPair}.
+   *
+   * @param keypair ed25519 keypair
+   */
+  constructor(private keypair: SignKeyPair) {}
+
+  /**
+   * Generate a new random keypair
+   */
+  static generate(): Keypair {
```

Comment:
Nit: The Rust homonym takes an RNG. I'm not sure if `nacl` has an equivalent interface that we could use optionally here or maybe use the Rust `new()` idiom?

Reply:
There is an internal abstraction that nacl uses to choose between the default browser or node.js random number generators depending on the current environment. Since nacl is always using each environment's defacto CSPRNG, I'm not sure we need to give that power to a user. They can always work around this randomly generating a seed by their own means and providing it to `Keypair.fromSeed`